controllers/apps/transformer_component_validation.go

```diff
@@ -21,14 +21,17 @@ package apps
 
 import (
 	"fmt"
+	"math"
 
 	appsv1alpha1 "github.com/apecloud/kubeblocks/apis/apps/v1alpha1"
 	"github.com/apecloud/kubeblocks/pkg/controller/graph"
 )
 
-const (
-	defaultMinReplicas         int32 = 0
-	defaultMinReplicas4ScaleIn int32 = 1
+var (
+	defaultReplicasLimit = appsv1alpha1.ReplicasLimit{
+		MinReplicas: 1,
```

Comment:
Default 0 replica is better than 1, especially for the autoscaling or serverless mode, a component is starting with replica=0 is reasonable.

Reply:
The intention is to ensure the following behaviors are consistent:

1. The default number of replicas is 1. If the component's spec.replicas is not specified, the replicas are set to 1.
2. The minimum number of replicas that can be scaled in is 1, unless the Addon specifies spec.replicasLimit.minReplicas to be 0 in cmpd.
3. However, if spec.replicasLimit.minReplicas is not set, replicas can be set to 0 during startup. This behavior appears inconsistent with point 2, so this PR fixes it.